### ISSUE TITLE
Support for multi-label classification (requires k-hot encoding)

### ISSUE THREAD

**Issue by BrianMiner** [aigamedev/scikit-neuralnetwork#86]
Fantastic work on this! I had one question and that was if it was possible to train a model with multiple labels as output. There are two cases I am thinking of:
 1) We have multiple labels and an object may be 0 or more of them. So I envision a single vector with all zeros, all ones or something in-between (mixed 1s and 0s)
2) There are multiple one hot vectors - each one denoting a different class. For example the first vector will be all zeroes except for the label of the instance, the next is likewise set up and so on for multiple one-hot vectors

If these are possible, what should the output data structure look like to work with the library?

**Comment by alexjc** [comment]
Thanks.

You could do this manually with a `Regressor` (and `Softmax` final layer) but the `Classifier` does not currently support this out-of-the-box.  It's a nice feature and I ran into this myself (discussed it with @ssamot), but I won't have time to do this in the short term.

Pull request would be welcome. I'd presume this could be done within the `Classifier` quite transparently?  Maybe sklearn has a standard approach for this!

**Comment by BrianMiner** [comment]
I found this for pylearn2 (for case #1 I mentioned). So its supported by 
the underlying software.
https://groups.google.com/d/msg/pylearn-users/0fIyp0QPiZ4/mY6c50lc7UAJ

**Comment by alexjc** [comment]
The thread seems to indicate you need to do this manually outside of PyLearn2. I will look into it when I have time. If you want to submit a PR it's welcome!

**Comment by alexjc** [comment]
You can try the `multi-label` branch to test this out, just use the standard classifier — should work transparently as long as you use `Sigmoid` as your last layer.

**Comment by BrianMiner** [comment]
Cool I will - what is the outcome structure, a k-hot encoded vector or a 
list of one-hot?

**Comment by alexjc** [comment]
The output is exactly the same as the training examples you feed it, for example a `numpy.ndarray` of (N, 3) with three classes of integers.
